app/controllers/users_controller.rb

```diff
@@ -37,9 +39,13 @@ def team_valid?
   end
 
   def setup_user_in_cognito
-    temporary_password = ('a'..'z').to_a.sample(3) + ('A'..'Z').to_a.sample(3) + ('0'..'9').to_a.sample(3) + ('!'..'/').to_a.sample(1)
+    temporary_password = ""
+    until password_meets_criteria?(temporary_password) do
```

Comment:
I think you can pass `temporary_password = ""` as a default value to save declaring it above

Reply:
or just default to nil in the method

`def password_meets_criteria?(password = nil)`
...
    `... password&.length ...`

Reply:
It turns out that won't work in this context (I think because it means the variable ends up being scoped to the loop, since that's where it will first get assigned), so we are stuck with the explicit declaration. :(